src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -152,54 +166,157 @@ object DepositHandler {
     }
   }
 
-  def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
-    log.debug(s"[$id] Checking if bag contains fetch.txt")
-    getFetchTxt(bagitDir).foreach(fetchText =>
-      { checkFetchTxtUrls(fetchText)
-        resolveFetchTxtFiles(bagitDir, fetchText)
-      }
-    )
-  }
-
-  private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
+  private def getFetchTxt(bagitDir: File): Option[FetchTxt] = Option {
     getBagFromDir(bagitDir).getFetchTxt
   }
 
-  private def checkFetchTxtUrls(fetchText: FetchTxt)(implicit id: String): Unit = {
+  def checkFetchItemUrls(bagitDir: File)(implicit id: String): Try[Unit] = Try {
     log.debug(s"[$id] Checking validity of urls in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      checkUrlValidity(item.getUrl))
+    getFetchTxt(bagitDir).foreach(fetchItem =>
+      fetchItem.asScala.foreach(item =>
+      checkUrlValidity(item.getUrl)))
   }
 
   private def checkUrlValidity(url: String)(implicit id: String): Unit = {
     // check if the url is syntactically correct
     try {
       new URL(url)
     } catch {
-      case e: MalformedURLException => throw new InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
+      case e: MalformedURLException => throw InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
     }
     // check if the url complies with the allowed url-structure
     val urlPattern = Pattern.compile(SwordProps("url-pattern"))
     if (!urlPattern.matcher(url).matches)
-      throw new InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
+      throw InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
   }
 
-  private def resolveFetchTxtFiles(bagitDir: File, fetchText: FetchTxt)(implicit id: String): Unit = {
-    log.debug(s"[$id] Resolving files in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      Using.urlInputStream(new URL(item.getUrl)).foreach(src =>
-        Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename))))
-  }
-
-  def checkBagValidity(bagitDir: File)(implicit id: String): Try[Unit] = {
+  def checkBagValidity(bagitDir: File)(implicit id: String, baseDir: File, baseUrl: URI) = {
     log.debug(s"[$id] Verifying bag validity")
+
+    val fetchItems = getFetchTxt(bagitDir).
```

Comment:
I noticed a slight confusion with our code conventions here. If you break up a sequence of operators into multiple lines, you put the `.` as the first character on the new line.

```scala
val fetchItems = getFetchTxt(bagitDir)
    .map(_.asScala)
    .getOrElse(Seq())
```

Also note the indent of the second and third line! In some cases you did not have that either.

Would you mind going through the code with this in mind?